suiteData.js

```diff
@@ -32,7 +38,19 @@ class SuiteData {
                 suiteData = await this.getSuiteData(request.query);
                 break;
             case "POST":
-                suiteData = new SuiteData(request.body, request.query);
+                let tests = request.body?.tests;
+                let suiteDataObj = request.body;
+                if (_.isString(tests)) {
+                    const testsDir = path.join(config.testsDir, sanitize(tests));
+                    if (await exists(testsDir)) {
+                        suiteDataObj.tests = (await listDir(testsDir))
```

Comment:
dude, it's by ref, it's the same as before...
what I meant, is something like this:
```
if (_.isString(tests)) {
    ...
    suiteData = new SuiteData((await ...).filter(...).map(...), request.query);
} esle {
    suiteData = new SuiteData(request.body, request.query);
}
}

Reply:
There are properties in the body other than "tests" that should be passed. We can create a new object and pick the relevant properties, but it's cumbersome. Since there is not additional middleware using the body I don't see any problem that the request's body itself is changed.